postgres.md

```diff
@@ -0,0 +1,404 @@
+Use uppercase for statements, operators and keywords
+```sql
+-- bad
+SELECT * FROM foo order by bar;
+
+-- good
+SELECT * FROM foo ORDER BY bar;
+```
+
+Use lowercase for functions
+```sql
+-- bad
+SELECT COUNT(*) FROM foo;
+
+-- good
+SELECT count(*) FROM foo;
+```
+
+Use lowercase for types
+```sql
+-- bad
+CREATE TABLE foo(bar INTEGER, baz TEXT);
+
+-- good
+CREATE TABLE foo(bar integer, baz text);
+```
+
+Do not use quotation marks for identifiers, unless you are defining a new type (which needs camelCase notation)
+```sql
+-- bad
+UPDATE "foo" SET "bar" = x;
+
+-- good
+UPDATE foo SET bar = x;
+
+-- good
+CREATE TYPE foo AS (
+  "fooId" integer,
+  "title" text,
+  "parentId" integer
+);
+```
+
+Use same difference operator as other languages.
+```sql
+-- bad
+SELECT baz FROM foo WHERE bar <> baz;
+
+-- good
+SELECT bar FROM foo WHERE bar != baz;
+```
+
+Use double pipe for explicit concatenation
+```sql
+-- bad
+SELECT 'foo'
+'bar';
+
+-- bad
+SELECT 'foobar';
+
+-- good
+SELECT 'foo' || 'bar';
+```
+
+Always specify the needed columns
+```sql
+-- bad
+SELECT * FROM foo;
+
+-- good
+SELECT foo_id, bar_id, creation_date FROM foo;
+```
+
+Use explicit parameters (with `in_` or `out_` prefix) instead of positional parameters
+```sql
+-- bad
+CREATE FUNCTION foo(text) RETURNS text AS $$
+  SELECT $1;
+$$ LANGUAGE sql;
+
+-- bad
+CREATE FUNCTION foo(bar text) RETURNS text AS $$
+  SELECT bar;
+$$ LANGUAGE sql;
+
+-- good
+CREATE FUNCTION foo(in_bar text) RETURNS text AS $$
+  SELECT in_bar;
+$$ LANGUAGE sql;
+```
+
+Use proper array declaration
+```sql
+-- bad
+SELECT '{1, 2, 3 + 4}';
+
+-- bad
+SELECT ARRAY[1,2,3+4];
+
+-- good
+SELECT array[1, 2, 3 + 4];

```

Comment:
In the construct `ARRAY[]`, `ARRAY` is not a type, it's a keyword; should be uppercase.